torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -23,8 +24,10 @@ const std::shared_ptr<torch::lazy::DimensionNode> DimCast(
 
 SizeNode::SizeNode(torch::lazy::Value input, size_t dim)
     : XlaNode(torch::lazy::OpKind{c10::Symbol::fromQualString("aten::size")},
-              {input}, xla::ShapeUtil::MakeShape(xla::S64, {}), 1,
-              torch::lazy::MHash(dim)),
+              {input},
+              xla::ShapeUtil::MakeShape(
```

Comment:
Just want to confirm that the change in this file is not directly related to the cast failure. And it is more of making the output type correct by making it depend on the input device type, right?

Reply:
yes, that's correct.